pkg/supportbundle/collect.go

```diff
@@ -207,136 +222,365 @@ func getAnalysisFile(analyzeResults []*analyze.AnalyzeResult) (io.Reader, error)
+	}
+	// Set up the ticker and timeout
+	ticker := time.NewTicker(100 * time.Millisecond) // Polling frequency
+	// timeout := constants.DEFAULT_LOGS_COLLECTOR_TIMEOUT
+	defer ticker.Stop()
+	// timeoutChan := time.After(timeout) // Constant timeout
+
+	// Loop until stdout has content or timeout is reached
+	for range ticker.C {
+		if stdout.Len() > 0 {
+			return stdout.Bytes(), stderr.Bytes(), nil
 		}
+	}
+	return stdout.Bytes(), stderr.Bytes(), nil
+}
 
-		// Send progress event: completed successfully
-		opts.ProgressChan <- fmt.Sprintf("[%s] Completed host collector", collector.Title())
+func runRemoteHostCollectors(ctx context.Context, hostCollectors []*troubleshootv1beta2.HostCollect, bundlePath string, opts SupportBundleCreateOpts) (map[string][]byte, error) {
+	output := collect.NewResult()
 
-		// Aggregate the results
-		for k, v := range result {
-			collectedData[k] = v
+	clientset, err := kubernetes.NewForConfig(opts.KubernetesRestConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: rbac check
+
+	nodeList, err := getNodeList(clientset, opts)
+	if err != nil {
+		return nil, err
+	}
+	klog.V(2).Infof("Node list to run remote host collectors: %s", nodeList.Nodes)
+
+	// create remote pod for each node
+	labels := map[string]string{
+		"troubleshoot.sh/remote-collector": "true",
+	}
+
+	var mu sync.Mutex
+	nodeLogs := make(map[string]map[string][]byte)
+
+	ds, err := createHostCollectorDS(ctx, clientset, labels)
+	if err != nil {
+		return nil, err
+	}
+
+	// wait for at least one pod to be scheduled
+	err = waitForDS(ctx, clientset, ds)
+	if err != nil {
+		return nil, err
+	}
+
+	klog.V(2).Infof("Created Remote Host Collector Daemonset %s", ds.Name)
+	pods, err := clientset.CoreV1().Pods(ds.Namespace).List(ctx, metav1.ListOptions{
+		LabelSelector:  selectorLabelKey + "=" + selectorLabelValue,
+		TimeoutSeconds: new(int64),
+		Limit:          0,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var eg errgroup.Group
+	for _, pod := range pods.Items {
+		eg.Go(func() error {
+			// TODO: set timeout waiting
+			if err := waitForPodRunning(ctx, clientset, &pod); err != nil {
+				return err
+			}
+
+			results := map[string][]byte{}
+			for _, collectorSpec := range hostCollectors {
+				// convert host collectors into a HostCollector spec
+				spec := createHostCollectorsSpec([]*troubleshootv1beta2.HostCollect{collectorSpec})
+				specJSON, err := json.Marshal(spec)
+				if err != nil {
+					return err
+				}
+				klog.V(2).Infof("HostCollector spec: %s", specJSON)
+
+				stdout, _, err := getExecOutputs(ctx, opts.KubernetesRestConfig, clientset, pod, specJSON)
+				if err != nil {
+					return err
+				}
+
+				result := map[string]string{}
+				if err := json.Unmarshal(stdout, &result); err != nil {
+					return err
+				}
+
+				for file, data := range result {
+					results[file] = []byte(data)
+				}
+
+				time.Sleep(1 * time.Second)
+			}
+
+			// wait for log stream to catch up
+			time.Sleep(1 * time.Second)
```

Comment:
Sleep isn't great here. Better to wait for the object. This will be brittle

Reply:
Thank you! Agree, I am trying to use polling method. However, the previous [PR](https://github.com/replicatedhq/troubleshoot/pull/1671) will be refactored. This PR will be on hold.